### PR TITLE
Drop support for PHP 7.3

### LINKED ISSUE #4434: Target PHP 7.4 for 4.0.x
I can see that `3.0.x` currently targets PHP 7.3+; could we make 7.4 the minimum requirement?

Reasons:

- PHP 7.3's active support [ends in 20 days](https://www.php.net/supported-versions.php) (6 December)
- Targeting 7.4 would allow us to:
  - harden the codebase a bit using typed properties
  - make it a bit cleaner by using short closures in a few places.

### PR DESCRIPTION
**Pull request by morozov**
|      Q       |   A
|------------- | -----------
| Type         | improvement
| BC Break     | no

Fixes #4434

### CODE REVIEW

**Review by derrabus — DISMISSED** [review]

File: src/Connection.php [hunk 1]
```diff
@@ -68,44 +68,32 @@ class Connection
 
     /**
      * The wrapped driver connection.
-     *
-     * @var \Doctrine\DBAL\Driver\Connection|null
      */
-    protected $_conn;
+    protected ?\Doctrine\DBAL\Driver\Connection $_conn = null;
```
Comment (SenseException) [inline src/Connection.php]: `use Doctrine\DBAL\Driver\Connection`

File: tests/Tools/Console/RunSqlCommandTest.php [hunk 2]
```diff
@@ -16,10 +16,8 @@
 
 class RunSqlCommandTest extends TestCase
 {
-    /** @var CommandTester */
-    private $commandTester;
-    /** @var RunSqlCommand */
-    private $command;
+    private CommandTester $commandTester;
+    private RunSqlCommand $command;
```
Comment (SenseException) [inline tests/Tools/Console/RunSqlCommandTest.php:20]:
```suggestion

    private RunSqlCommand $command;
```

**Review by SenseException — DISMISSED** [review]

File: src/Connection.php (hunk 1 above)
Comment (derrabus) [reply src/Connection.php]: … `as DriverConnection` (this file already declares a class of that name.)

**Review by greg0ire — APPROVED** [review]
Looks OK to me. I'm not to sure about doing such things now on this branch though: it causes a big divergence between branches, so merges up might become more difficult. By the time 4.0 is released, I think 7.3 will have been dropped from 3.x too, won't it?

**Comment by derrabus** [comment]
@greg0ire We're applying pretty much the same changes to Symfony's 6.0 branch right now. Yes, they do cause merge conflicts, but they're easy to fix in most cases.

**Comment by morozov** [comment]
> I'm not to sure about doing such things now on this branch though: it causes a big divergence between branches, so merges up might become more difficult.

Most of the changes in this PR are automated. So if there are new conflicts caused by the property types, they should be possible to resolve in favor of 3.x and then fixed by `phpcbf`. Doesn't look like a big deal to me.

> By the time 4.0 is released, I think 7.3 will have been dropped from 3.x too, won't it?

I don't see much point in doing it. Specifically, I don't see a point in introducing property types in `3.x`: compared to `4.0.x`, most of the methods don't have scalar argument/return types enforced, enforcing them at the property level will lead to a nightmare.

**Comment by greg0ire** [comment]
Thanks to you both! What's wrong with the build?

**Comment by morozov** [comment]
> What's wrong with the build?

I'll disable the PHP 7.3 jobs right before the merge. UPD: there aren't actually other PRs against 4.0.x, so we can safely disable them now.

**Pull request merged**